notebook/notebookapp.py

```diff
@@ -1805,7 +1833,13 @@ def launch_browser(self):
         if not browser:
             return
 
-        if self.file_to_run:
+        if not self.use_redirect_file:
+            uri = self.default_url[len(self.base_url):]
+
+            if self.token:
+                uri = url_concat(uri, {'token': self.token})
+
+        elif self.file_to_run:
```

Comment:
`use_redirect_file` and `file_to_run` are orthogonal, so they can't be in an if/else. This would mean that `jupyter notebook foo.ipynb` wouldn't open to `foo.ipynb` if you've set `use_redirect_file = False`.

Reply:
Okay. I was thinking this was the redirect `file_to_run`, not an argument file.